utils/utils.py

```diff
@@ -62,36 +62,7 @@ def load_from_checkpoint(checkpoint, model, optimizer=None, inference=False):
         model: model to replace
         optimizer: optimiser to be used
     """
-    # Corrects exception with test loop. Problem with loading generic checkpoint into DataParallel model
-    # https://github.com/bearpaw/pytorch-classification/issues/27
-    # https://discuss.pytorch.org/t/solved-keyerror-unexpected-key-module-encoder-embedding-weight-in-state-dict/1686/3
-    if isinstance(model, nn.DataParallel) and not list(checkpoint['model'].keys())[0].startswith('module'):
-        new_state_dict = model.state_dict().copy()
-        new_state_dict['model'] = {'module.'+k: v for k, v in checkpoint['model'].items()}    # Very flimsy
-        del checkpoint
-        checkpoint = {}
-        checkpoint['model'] = new_state_dict['model']
```

Comment:
These lines were useful for a specific use case: training saves a generic non-DataParallel model. Then we load that model during test time into an instanciated DataParallel model. Throws exception if key in pretrained non-DataParallel dict don't start with "module.". I have a feeling these lines should stay, but I'm willing to try without, then revert if necessary.